roles/libvirt_manager/tasks/manage_vms.yml

```diff
@@ -70,13 +70,14 @@
     - vm_type is not match('^(crc|ocp).*$')
   delegate_to: "{{ vm }}.{{ inventory_hostname }}"
   remote_user: "{{ _init_admin_user }}"
+  vars:
+    _fqdn: "{{ vm }}.{{ cifmw_reproducer_domain | default('local') }}"
   ansible.builtin.shell:
     executable: /bin/bash
     cmd: |-
       test -d /home/zuul && exit 0;
       set -xe -o pipefail;
-      echo "{{ vm }}" | sudo tee /etc/hostname;
-      sudo hostname -F /etc/hostname;
+      sudo hostnamectl hostname "{{ _fqdn }}"
```

Comment:
Yes, let `hostnamectl` set it.

Reply:
the canonical hostname of the edpm node if its hosting a nova is required to be the ctrlplane fqdn

so cifmw_reproducer_domain must always equal the ctlplane network domain in the network atachmetn definition.

without this it will break live migration and tls and it will cause issues with adoption potentially too.

Reply:
If you set hostname with fqdn, and use the same in nodeset spec node `hostName`, it would be used as canonical_hostname and resolve to the ctlplane network ip. I guess you're not updating the nodeset spec with the correct `hostName`.